core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -51,6 +54,15 @@
 
     private static final Logger LOGGER = Logger.getLogger(AtomicFileWriter.class.getName());
 
+    private static /* final */ boolean LOSE_INTEGRITY = SystemProperties.getBoolean(
+            AtomicFileWriter.class.getName() + ".LOSE_DATA_INTEGRITY_FOR_PERFORMANCE");
+
+    static {
+        if (LOSE_INTEGRITY) {
+            LOGGER.log(Level.SEVERE, "LOSE_DATA_INTEGRITY_FOR_PERFORMANCE flag used, YOU RISK LOSING DATA.");
```

Comment:
Again, overly dramatic.  I think we really only need to warn people about this case happening when the OS or storage is yanked rather than cleanly shut down.